src/main/java/org/kuali/kra/budget/document/BudgetDocumentRule.java

```diff
@@ -154,7 +154,86 @@ protected boolean processCustomSaveDocumentBusinessRules(Document document) {
     * @param budgetDocument
     * @return
     */
-    @SuppressWarnings("rawtypes")
+    @SuppressWarnings("rawtypes")  
+	protected boolean processBudgetProjectIncomeBusinessRule(BudgetDocument budgetDocument) {
+        boolean valid = true;
+        MessageMap errorMap = GlobalVariables.getMessageMap();
+        int i = 0;
+        
+        // Get each budget cost share.
+        for (BudgetCostShare budgetCostShare : budgetDocument.getBudget().getBudgetCostShares()) {
```

Comment:
is it possible that getBudget() can come back null, or that a budget.getBudgetCostShares()? (Honest question, I'm having issues w/ kc_project, so can't look right now). If so, that can be checked, and made another validation here. Also, of course would be bad to throw an NPE.